### PR TITLE
light/provider/http: fix Validators

### LINKED ISSUE #6010: light client cannot work when the number of validators is equal 100
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source):

unreleased-master-b1646e51e263d579b4cc522668b4a2b81c1f56bd(v0.34.*)

**What happened**:

If the number of validators is equal `100`(defined by [maxPerPage](https://github.com/tendermint/tendermint/blob/master/light/provider/http/http.go#L97)), the light client cannot work.

**How to reproduce it**

- Option 1: Modify  `maxPerPage` to `1` and run a network with one validator, https://github.com/chengwenxi/tendermint/tree/vincent-maxPerPage-1

  1. run a testnet network

     ```bash
     tendermint testnet --v 1
     tendermint node --home mytestnet/node0/ --proxy-app kvstore
     ```

  2. run a light client proxy server

     ```bash
     tendermint light chain-98I3B2 -p http://localhost:26657 --height 102 --hash 68778AD7342039B2DC46047B8221DAECCE0E2F663AEA16EFD51F94BF7A72DF26 -w http://localhost:26657
     ```

  It will be stuck for a long time and then return the following error:

  ```
  I[2021-01-29|16:16:45.341] Removing all light blocks                    
  I[2021-01-29|16:16:45.347] Downloading trusted light block using options 
  ERROR: client failed to respond. Tried to replace primary but: no witnesses connected. please reset light client
  ```

- Option 2: Use an existing network with 100 validators. The IRISnet bifrost testnet that based on tendermint@v0.34.1 and the validators has reached 100, we can access it by public node `34.80.22.255:26657`.

  ```bash
  tendermint light bifrost-2 -p http://34.80.22.255:26657 --height 140150 --hash 0401E3281401DEFBFCCAF650B98A950282A7043AC5962331ECD02ADD8A2D56F5 -w http://34.80.22.255:26657
  ```

  It returns the same error as ` Option 1`.

**Why happened**:

In function [validatorSet](https://github.com/tendermint/tendermint/blob/master/light/provider/http/http.go#L102-L117),
we retry call `p.client.Validators` to get all validators by pages, if the `len(validatorSet) = maxPerPage`, `p.client.Validators` will return `page should be within [1, {total_page}] range, ​{page}`, see https://github.com/tendermint/tendermint/blob/master/rpc/core/env.go#L111.

**Proposal**

Minor fix: ignore this error in function validatorSet.

### PR DESCRIPTION
**Pull request by melekes**
Closes #6010

### CODE REVIEW

File: light/provider/http/http.go [hunk 1]
```diff
@@ -93,15 +95,21 @@ func (p *http) ReportEvidence(ctx context.Context, ev types.Evidence) error {
...
-		vals       = []*types.Validator{}
-		page       = 1
+		perPage = 100
+		vals    = []*types.Validator{}
+		page    = 1
+		total   = -1
 	)
 
-	for len(vals)%maxPerPage == 0 {
+	for len(vals) != total && page < maxPages {
```
Comment (chengwenxi) [inline light/provider/http/http.go:110]: `page <= maxPages`?

**Review by cmwaters — APPROVED** [review]
Nice work

**Comment by colin-axner** [comment]
Will this be included in 0.34.4? Is there a place I can track this?

**Comment by cmwaters** [comment]
Yup we should backport this (@melekes)

**Pull request merged**